pyrtl/rtllib/multipliers.py

```diff
@@ -155,6 +155,44 @@ def tree_multiplier(A, B, reducer=adders.wallace_reducer, adder_func=adders.kogg
     return reducer(bits, bits_length, adder_func)
 
 
+def signed_tree_multiplier(A, B, reducer=adders.wallace_reducer, adder_func=adders.kogge_stone):
+    """Same as tree_multiplier, but uses two's complement signed integers"""
+    if len(A) == 1 or len(B) == 1:
+        raise pyrtl.PyrtlError("sign bit required, one or both wires too small")
+
+    aneg = A[-1]
+    bneg = B[-1]
+    a = pyrtl.WireVector(len(A))
+    b = pyrtl.WireVector(len(B))
+    neg = aneg ^ bneg
+    with pyrtl.conditional_assignment:
+        with aneg:
+            a |= ~A + 1
+        with pyrtl.otherwise:
+            a |= A
```

Comment:
You're duplicating the same 5 lines quite a lot, Could you make that a private function?